File: 0001-two-sum/0001-two-sum.cpp
class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> record;
            vector<int> result;

            for(int i=0; i<nums.size(); i++){
                int diff = target - nums[i];
                if(record.find(diff) != record.end()){
                    result.push_back(record[diff]);
                    result.push_back(i);
                }
                record[nums[i]] = i;

            }

            return result;
        }
};

File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $hashMap = [];

        for($i = 0; $i < count($nums); $i++){
            if (isset($hashMap[$target - $nums[$i]])) {
                return [$i, $hashMap[$target - $nums[$i]]];
            }
            $hashMap[$nums[$i]] = $i;
        }

        return [];
    }
}